datamade/la-metro-councilmatic
# Add server-side querying of SmartLogic API

## Overview

This PR adds a pass-through route to query SmartLogic. Implementing this server-side allows us to keep our API credentials from being exposed in client-side code. 
 
### Checklist


### Notes

Some APIs, like Google Maps, include options to restrict requests by domain, so it is not a risk to expose keys in the source code. The SmartLogic API does not include this option, so in consultation with them, we have identified a server-side implementation as the best course of action. If this implementation proves too slow, the next-best option is to build an endpoint that exposes the expiring token, and do the querying in JavaScript, client-side.

## Testing Instructions

- Add the new SmartLogic values to your `settings_deployment.py` file. (Ping me for the API key.)
- Go to the homepage and try typing a value. Confirm that suggestions are auto-populated as you type.

Related to #433.
